conda_smithy/azure_ci_utils.py

```diff
@@ -189,16 +191,32 @@ def register_repo(github_org, repo_name, project_id=AZURE_PROJECT_ID):
         )
 
 
-def repo_registered(github_org, repo_name, project_id=AZURE_PROJECT_ID):
-    from vsts.build.v4_1.models import BuildDefinitionReference
-    from vsts.build.v4_1.build_client import BuildClient
-
-    bclient: BuildClient = connection.get_client(
+def build_client() -> BuildClient:
+    return connection.get_client(
         "vsts.build.v4_1.build_client.BuildClient"
     )
 
+
+def repo_registered(github_org, repo_name, project_id=AZURE_PROJECT_ID):
     existing_definitions: typing.List[
         BuildDefinitionReference
-    ] = bclient.get_definitions(project=project_id, name=repo_name)
+    ] = build_client().get_definitions(project=project_id, name=repo_name)
 
     return bool(existing_definitions)
+
+
+def enable_reporting(user, repo, project_id=AZURE_PROJECT_ID):
+    bclient = build_client()
+    bdef_header = bclient.get_definitions(project=project_id, name=repo)[0]
+    bdef = bclient.get_definition(bdef_header.id, bdef_header.project.name)
+    bdef.repository.properties['reportBuildStatus'] = 'true'
+    bclient.update_definition(bdef, bdef.id, bdef.project.name)
+
+
+def get_build_id(user, repo, project_id=AZURE_PROJECT_ID):
```

Comment:
Can we set this up in conda-forge-webservices and have a link there that redirects to azure?

Reply:
Plan is to have conda-forge-webservices have an azure-token so that as part of a rerender it can acquire this id.
This can also be used to do things like enable azure-reporting via the bot.

Reply:
Do you need a azure token to do this? It can be done without a token.

See also https://github.com/Microsoft/azure-pipelines-agent/issues/1656#issuecomment-468690515

Reply:
So i have found a way to obtain the build_id without using a token.  This should be fine unless we rename repos.

EG:
https://dev.azure.com/conda-forge/feedstock-builds/_apis/build/definitions?name=zlib-feedstock 
works without authentication and gives you all the pieces you need.

Reply:
We have 2 options.

1. Get the definition id from that url in each rerender.
2. Wait for the API without definition id

I'm fine with either.

Reply:
We could always store the id as well and not hit it each rerender.

Reply:
Or we just do both 😄